test/system/projects/samples/attachments_test.rb

```diff
@@ -18,9 +19,9 @@ class AttachmentsTest < ApplicationSystemTestCase
 
       test 'user with role >= Maintainer should be able to see empty state with upload message' do
         visit namespace_project_sample_url(@namespace, @project, @sample2)
-        assert_selector 'a', text: I18n.t('projects.samples.show.new_attachment_button'), count: 1
-        assert_no_selector 'button', text: I18n.t('projects.samples.show.concatenate_button')
-        assert_no_selector 'button', text: I18n.t('projects.samples.show.delete_files_button')
+        assert_selector 'a', text: I18n.t('projects.samples.show.new_attachment_button')
+        assert_selector 'button', text: I18n.t('projects.samples.show.concatenate_button'), visible: false
```

Comment:
We can assert if the button is disabled using `assert_selector 'button[disabled]', text: BUTTON_TEXT_HERE`. These 2 assertions can be updated to use these assertions. The `visible` here currently implies that its not visible on the page which it currently is